integrations/github/src/misc/entity-mapping.ts

```diff
@@ -0,0 +1,252 @@
+import * as sdk from '@botpress/sdk'
+import {
+  User as GitHubUser,
+  Issue as GitHubIssue,
+  PullRequest as GitHubPullRequest,
+  Discussion as GitHubDiscussion,
+  Repository as GitHubRepository,
+  Label as GitHubLabel,
+  PullRequestReview as GitHubPullRequestReview,
+} from '@octokit/webhooks-types'
+
+import { User, Issue, PullRequest, Discussion, Repository, Label, PullRequestReview } from 'src/definitions/entities'
+import * as bp from '.botpress'
+
+abstract class BaseEntityMapper<G extends object, B extends object> {
+  protected abstract map(_entity: G, ..._args: unknown[]): Promise<B>
+
+  public async mapEach(entities: G[]): Promise<B[]> {
+    return Promise.all(entities.map((entity) => this.tryMap(entity)))
+  }
+
+  public async tryMap(entity: G, ...args: unknown[]): Promise<B> {
+    try {
+      return await this.map(entity, ...args)
+    } catch (thrown: unknown) {
+      throw new sdk.RuntimeError(`Failed to map entity: ${thrown}`)
+    }
+  }
+}
+
+class UserEntityMapper extends BaseEntityMapper<GitHubUser, User> {
+  public constructor(protected readonly client: bp.Client) {
+    super()
+  }
+
+  protected async map(githubUser: GitHubUser): Promise<User> {
+    return {
+      name: githubUser.name ?? githubUser.login,
+      handle: githubUser.login,
+      id: githubUser.id,
+      nodeId: githubUser.node_id,
+      url: githubUser.html_url,
+      botpressUser: (await this._getOrCreateBotpressUserFromGithubUser({ githubUser })).id,
+    }
+  }
+
+  private async _getOrCreateBotpressUserFromGithubUser({ githubUser }: { githubUser: GitHubUser }) {
+    const { users } = await this.client.listUsers({
```

Comment:
**question:** why not `client.getOrCreate()` ?

Reply:
We could also add the unique constraint directly to the integration definition to avoid having to specify it every time. For instance:

```typescript
export default new IntegrationDefinition({
  user: {
    tags: {
      handle: {},
      nodeId: {},
    },
    uniqueTags: ['nodeId'],
  }
})
```